std/json.d

```diff
@@ -359,6 +359,80 @@ struct JSONValue
         return type == JSONType.null_;
     }
 
+    /***
+     * Generic type value getter
+     * A convenience getter that will try to convert the underlaying data to the given type.
+     * Note: only arithmetic, `bool`, `string`, `JSONValue` and `JSONValue[]` types are accepted
+     * Throws: `JSONException` if `T` is not a valid type or conversion is impossible
+     */
+    @property inout(T) get(T)() inout const pure @safe
+    {
+        static if (is(T : string))
+        {
+            return str;
+        }
+        else static if (is(T : bool))
+        {
+            return boolean;
+        }
+        else static if (__traits(isArithmetic, T))
+        {
+            switch (type)
+            {
+            case JSONType.float_:
+                return cast(T)floating;
+            case JSONType.uinteger:
+                return cast(T)uinteger;
+            case JSONType.integer:
+                return cast(T)integer;
+            default:
+                throw new JSONException("JSONValue is not a number type");
+            }
+        }
+        else
+        {
+            throw new JSONException("Unsupported type");
+        }
+    }
+    // This specialization is needed because arrayNoRef requires inout
+    @property inout(T) get(T : JSONValue[])() inout pure @trusted
+    {
+        return arrayNoRef;
+    }
+    // ditto
+    @property inout(T) get(T : JSONValue[string])() inout pure @trusted
+    {
+        return object;
+    }
+    ///
+    @safe unittest
+    {
+        import std.exception;
+        string s =
+        `{
+            "a": 123,
+            "b": 3.1415,
+            "c": "text",
+            "d": true,
+            "e": [1, 2, 3],
+            "f": { "a": 1 }
+         }`;
+
+        immutable auto json = parseJSON(s);
+        assert(json["a"].get!double == 123.0);
+        assert(json["a"].get!int == 123);
+        assert(json["b"].get!double == 3.1415);
+        assert(json["b"].get!int == 3);
+        assert(json["c"].get!string == "text");
+        assert(json["d"].get!bool == true);
+        assertNotThrown(json["e"].get!(JSONValue[]));
+        assertNotThrown(json["f"].get!(JSONValue[string]));
+        assertThrown(json["a"].get!JSONException);
+        assertThrown(json["e"].get!string);
+        assertThrown(json["d"].get!double);
```

Comment:
Ironically `double d = true` works in D, but probably we should avoid perpetuating that mistake.

Reply:
OK, I will change that. Thank you!

Reply:
Sorry for being vague. I think no change is required here.